velox/connectors/hive/storage_adapters/abfs/AbfsFileSystem.cpp

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "velox/connectors/hive/storage_adapters/abfs/AbfsFileSystem.h"
+#include "velox/common/file/File.h"
+#include "velox/connectors/hive/HiveConfig.h"
+#include "velox/connectors/hive/storage_adapters/abfs/AbfsReadFile.h"
+#include "velox/connectors/hive/storage_adapters/abfs/AbfsUtil.h"
+#include "velox/core/Config.h"
+
+#include <fmt/format.h>
+#include <folly/executors/IOThreadPoolExecutor.h>
+#include <glog/logging.h>
+
+namespace facebook::velox::filesystems::abfs {
+class AbfsConfig {
+ public:
+  AbfsConfig(const Config* config) : config_(config) {}
+
+  std::string connectionString(const std::string& path) const {
+    if (!connectionString().empty()) {
+      // This is only used for testing.
+      return connectionString();
+    }
+    auto abfsAccount = AbfsAccount(path);
+    auto key = abfsAccount.credKey();
+    VELOX_USER_CHECK(
+        config_->isValueExists(key), "Failed to find storage credentials");
+
+    return abfsAccount.connectionString(config_->get(key).value());
+  }
+
+  std::string connectionString() const {
+    return config_->get<std::string>(
+        AbfsFileSystem::kReadAbfsConnectionStr, "");
+  }
+
+ private:
+  const Config* FOLLY_NONNULL config_;
+};
+
+class AbfsFileSystem::Impl {
+ public:
+  explicit Impl(const Config* config) : abfsConfig_(config) {
+    LOG(INFO) << "Init ABFS file system";
```

Comment:
hey @majetideepak thank you very much for the review！

just back from china holiday.

currently the read/write implement only guarantee supporting azure storage abfs/abfss schema. So keep current msg should be fine, since the people who use Azure Blob/Gen2 storage should know what it means.

We can refactor the msg futher if also plan extend to support wasb schema for w/r.

Reply:
@gaoyangxiaozhu thanks for the clarification. The naming confuses me since the ABFS abbreviation already includes File System. We must remove the redundant `file system` suffix from the `ABFS file system`.

I also see [here](https://learn.microsoft.com/en-us/azure/storage/blobs/data-lake-storage-abfs-driver) that ABFS(Azure Blob Filesystem driver) is the new driver for the Data Lake Storage Gen2 suited for big data analytics. The older driver WASB is not as performant. Is there a use case in the future where we might have to support this?

Can you also add some comments in the code [here](https://github.com/facebookincubator/velox/pull/6675/files#diff-c72733576fc3011a5b0e045cfa86c572ae550af8482e2638a9158e99bb58d528R22) above the `class AbfsFileSystem` about the current support for `abfs, abfss` and the difference between the two as well a note on wasb?

Reply:
hey, @majetideepak 

Firstly, thank you for heading up the ABFS abbreviation issue. You are right, ABFS abbreviation already includes File system, so we need remove redundant file system suffic or renaming class name to avoid redundant.

Secondly, yes, abfs(s) are new driver/schema identifers that azure storage provided to integrate with azure blob storage with better performance and filesystem semantics support. And for your question, actually I think there is no need support wasb schema. 
Before I thought we may need support WASB because of compatible problem if customer try to use velox in thier old platform which still stuck using wasb, but actually  the right way if customer want to integrate azure blob with velox lib is upgrading their platform to use abfs.

In summary, I think we can only condiser support access Azure Blob Storage via abfs(s) schema,  wasb(s) is not supported since [Azure storage team allready claim it has been deprecated](https://learn.microsoft.com/en-us/azure/databricks/storage/azure-storage).

And for avoiding redundant name issue, I have made update to change msg part to `Init/Dispose Azure Blob file system` following your suggestion.

I was thinking if it is also worth to rename `AbfsFileSystem` class to `ABSFileSystem` class, ABS means (Azure Blob Storage) to better avoid redundant naming issue ?
If you agree the class rename part change, i will do the change or we can still keep current class name.

thank you again for the review.

Reply:
We have a similar naming class (HdfsFileSystem) for HDFS as well. This is fine.